internal/warn_on_deprecated_api.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+// deno-lint-ignore no-explicit-any
+const { Deno } = globalThis as any;
+
+const ALREADY_WARNED_DEPRECATED = new Set<string>();
+
+interface WarnDeprecatedApiConfig {
+  /** The name of the deprecated API. */
+  apiName: string;
+  /** The stack trace of the deprecated API. */
+  stack: string;
+  /** The version in which the API will be removed. */
+  removalVersion: string;
+  /** An optional message to print. */
+  suggestion?: string;
+}
+
+/**
+ * Prints a warning message to the console for the given deprecated API.
+ *
+ * These warnings can be disabled by setting `NO_DEPRECATION_WARNINGS=1`
+ * in the current process.
+ */
+export function warnOnDeprecatedApi(config: WarnDeprecatedApiConfig) {
+  const variable = "NO_DEPRECATION_WARNINGS";
+  if (
+    Deno?.permissions.querySync({ name: "env", variable }).state ===
```

Comment:
It seems Deploy doesn't have `querySync`.

Also how about checking this outside of `warnOnDeprecatedApi` to reduce the overhead?

Reply:
Please open a bug for the missing Deploy feature

Reply:
Created

Reply:
I've gone with `Deno.permissions.query()`, which is implemented in Deploy.